app/src/main/java/de/tum/in/tumcampusapp/cards/NextLectureCard.java

```diff
@@ -162,28 +163,27 @@ protected Notification fillNotification(NotificationCompat.Builder notificationB
         return notificationBuilder.build();
     }
 
-    public void setLectures(Cursor cur) {
-        do {
+    public void setLectures(List<de.tum.in.tumcampusapp.models.tumo.CalendarItem> calendarItems) {
+        for (de.tum.in.tumcampusapp.models.tumo.CalendarItem calendarItem: calendarItems) {
             CalendarItem item = new CalendarItem();
-            item.start = Utils.getDateTime(cur.getString(1));
-            item.end = Utils.getDateTime(cur.getString(2));
+            item.start = Utils.getDateTime(calendarItem.getDtstart());
+            item.end = Utils.getDateTime(calendarItem.getDtend());
 
             // Extract course title
-            item.title = cur.getString(0);
+            item.title = calendarItem.getTitle();
             item.title = item.title.replaceAll("[A-Z, 0-9(LV\\.Nr)=]+$", "");
             item.title = item.title.replaceAll("\\([A-Z]+[0-9]+\\)", "");
             item.title = item.title.replaceAll("\\[[A-Z]+[0-9]+\\]", "");
             item.title = item.title.trim();
```

Comment:
Also: this CalendarItem to CalendarItem logic probably fits better into a member function of CalendarItem